docs/spec.md

```diff
@@ -1923,6 +1923,55 @@ imaginary values, `lhs` and `rhs`, and produces a `result` tensor.
 
 &nbsp;[More Examples](../stablehlo/tests/interpret/complex.mlir)
 
+### composite
+
+#### Semantics
+
+Encapsulates an operation made up (composed) of other StableHLO operations,
+taking `inputs` and producing `results`. The decomposition is specified
+using the `decomposition` attribute. The composite op can be replaced with
+its decomposition without changing semantics.
+
+The `version` and `decomposition_attributes` enable additional information
+to be communicated between producers and consumers. `version` defaults to 0
+and `decomposition_attributes` defaults to an empty dictionary.
+
+#### Inputs
+
+| Label | Name                                   | Type                                          |
+|-------|----------------------------------------|-----------------------------------------------|
+| (I1)  | `inputs`                               | variadic number of values                     |
+| (I2)  | `name`                                 | constant of type `string`                     |
+| (I3)  | `decomposition`                        | constant of type `function`                   |
+| (I4)  | `version`                              | constant of type `si32`                       |
+| (I5)  | `decomposition_attributes`             | attribute dictionary                          |
+
+#### Outputs
+
+| Name      | Type                      |
+|-----------|---------------------------|
+| `results` | variadic number of values |
+
+#### Constraints
+
+* (C1) `re.fullmatch(r'[a-zA-Z]+(\.[a-zA-Z][a-zA-Z0-9_]*)+', name)`
```

Comment:
Let's abstract this to a helper method `is_namespaced_op_name(name)`. Let's also support underscore after dot. Some existing op names in the TF world that should be considered valid:
- `tf_executor._SwitchN
- `tf._XlaHostComputeMlir`

I can also see a use case for `dialect_v1.my_op`.

How about:
- Starts with letter
- [letter, number, underscore]*
- (`.` [letter, number, underscore, `$`]+ )+

`$` is used in some custom calls as target name, so I'm hesitant to suggest changes there, in case we want to support cc target + decomposition